lighthouse/aggregate-service-import.md

```diff
@@ -0,0 +1,137 @@
+- Mark conflicting `ServiceExport` objects with the `ServiceExportConflict` condition
+
+There are changes required in Lighthouse's handling of `EndpointSlices` before single
+`ServiceImports` can be introduced.
+
+- The `ServiceImports` for an exported service each contain the service VIP from the exporting
+  cluster. `ServiceImports` cannot be merged like this because a `ServiceImport` cannot have
+  more than one IP. The semantics of the IP are that it should be a cluster local Service VIP,
+  if needed. The service VIPs from the exporting clusters should be stored in `EndpointSlices`,
+  with a separate `EndpointSlice` per exporting cluster.
+
+It is worth noting that:
+
+- Headless services already have `EndpointSlices` containing the correct pod IP addresses, and contain
+  no VIP information in the multiple `ServiceImports`. A merged  `ServiceImport` for headless
+  services would comprise a merge of the port definitions in those `ServiceImports`.
+
+- The current GlobalNet implementation exports `EndpointSlices` for normal services that contain
+  pod IP addresses from the originating cluster.
+
+## Proposal
+
+Submariner has a passive broker, it is a central apiserver that is accessed by each cluster in
+the clusterset. Clusters synchronise resources to and from the broker with all sync and
+reconciliation logic implemented in the clusters.
+
+In this proposal we retain the passive broker and take an approach to merging `ServiceImports`
+where exporting clusters handle the merging. Simplistically, a merged `ServiceImport` outcome
+can be achieved by allowing the first exporting cluster to create a `ServiceImport` and all
+subsequent exporting clusters to reuse it, assuming they have matching port lists.
+
+1. First exporting cluster creates a `ServiceImport` on the broker.
+1. Subsequent exporting clusters reuse the existing `ServiceImport`.
+1. Exporting clusters create and sync `EndpointSlices` to the broker with the service VIP and a
+   `multicluster.kubernetes.io/service-name` label that matches the `ServiceImport`.
+1. All clusters sync the `ServiceImport` and `EndpointSlices` from the broker.
+
+Looking at the `ServiceImport` merging in more detail, the MCS specification asks for all port
+definitions from the `ServiceExports` to be combined in the merged `ServiceImport`. This means
+that an exporting cluster would need to merge its port definitions into the `ServiceImport`.
+Conversely, when removing a `ServiceExport` any merged port definitions would need to be
+removed. However an exporting cluster only has visibility of its local `ServiceExport` so cannot
+unmerge with the information available locally.
+
+Note: It is an open question what behaviour is expected when a connection request for service
+port is directed to a cluster that does not expose that port.
+
+The notion that services are equivalent when they have matching name and namespace but different
```

Comment:
Can we make it a requirement/limitation that services must have same name, namespace *and* matching port definitions? Mismatch in ports also results in a ServiceExport conflict. Maybe add a Design Limitation section and explain it all there, alongwith a comment on how it aligns with MCS API [as in implementation choice].

Reply:
Good suggestion – I will add an explicit requirement that service equivalence will be same name, namespace and port definitions. I will add a design limitation section.